yellowbrick/classifier/class_prediction_error.py

```diff
@@ -167,8 +152,6 @@ def finalize(self, **kwargs):
         cmax = max([sum(predictions) for predictions in self.predictions_])
         self.ax.set_ylim(0, cmax + cmax * 0.1)
 
-        # Put the legend outside of the graph
-        plt.legend(bbox_to_anchor=(1.04, 0.5), loc="center left")
         plt.tight_layout(rect=[0, 0, 0.85, 1])
```

Comment:
I will find a substitute (if it's there).

Reply:
@bbengfort I tried to find an alternative to plt.tight_layout and came across one possible solution. We can set the axes position manually(using `ax.set_position`). But this may fail for different number subplots.
Also plt.tight_layout works on subplots (i.e. it makes sure that labels of subplots do not intersect). So having it does no harm. It my point of view and I would like to know what you think of this. Also let me know if you have some ax function on your mind that can handle this.

Reply:
So I messed around with this some and I'm not terribly happy with what's happening. For example, the `0.85` box width you specified is very large, if the class labels are `[A, B, C]`, then this is far too much and a box width of `0.95` is sufficient to have the legend on the figure nicely. If the labels are `["hobbies", "books", "cinema"]` -- then you would want `0.85` or maybe even `0.75`. Magic numbers like these and global variables are usually warning signs. 

Ideally, we'd want something like option 4 as described in [this StackOverflow answer](https://stackoverflow.com/questions/4700614/how-to-put-the-legend-out-of-the-plot#answer-43439132) - where the bounding box is automatically computed based on the legend size. However, I did try playing around with it and it's not exactly what we want either. We could try adding dedicated legend axes, but then we go down the road of the problems that we're having in the JointPlot visualizer. 

Bottom line is that this is not the problem we're trying to solve in this PR - so if you could make an issue for it, we'll have to get to it in the future! Thank you for researching it!

Reply:
Sure, I will make an issue out of it. But I had a question. By changing box width the plot size becomes smaller and that needs to be handled, right?

Reply:
That's a good question - I experimented some more, it looks like the figure size is fixed, so changing the box width is more of an aspect ratio thing, the subplot is backing off that percentage and pulling the legend with it. More research required I think.